### PR TITLE
Delete comment feature

### LINKED ISSUE #84: As a careless avenger, I can delete comments on student's assignments
As a careless avenger, I can delete comments on student's assignments in case I accidentally comment on one.

### PR DESCRIPTION
**Pull request by Originalidk**
Add deletecomment feature with test cases.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommentCommand.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        if (!AssignmentName.isValidName(this.assignmentName.toString())) {
+            throw new CommandException(AssignmentName.MESSAGE_CONSTRAINTS);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/commands/DeleteCommentCommand.java:64]: test case

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -115,6 +117,12 @@ public Command parseCommand(String userInput) throws ParseException {
         case DeleteGradeCommand.COMMAND_WORD:
             return new DeleteGradeCommandParser().parse(arguments);
 
+        case EditCommentCommand.COMMAND_WORD:
+            return new EditCommentCommandParser().parse(arguments);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:124]: test case

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 3]
```diff
@@ -115,6 +117,12 @@ public Command parseCommand(String userInput) throws ParseException {
...
+
+        case DeleteCommentCommand.COMMAND_WORD:
+            return new DeleteCommentCommandParser().parse(arguments);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:127]: test case

**Review by ldinghan — COMMENTED** [review]
lgtm

**Review by wesho1107 — COMMENTED** [review]
Other than the nits pointed out by others, LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteCommentCommand.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        if (!AssignmentName.isValidName(this.assignmentName.toString())) {
```
Comment (Respirayson) [inline src/main/java/seedu/address/logic/commands/DeleteCommentCommand.java]: perhaps you want to change the parameters of `isValidName` to accept a AssignmentName instead? if not it seems like you're circling around by converting it into a String and then back into AssignmentName in the method

**Review by Respirayson — APPROVED** [review]
small minor things only but looks good

**Pull request merged**